### PR TITLE
JENA-2238: RowSet readers and writers

### PR DESCRIPTION
**Pull request by afs**
Refactoring so that ResultSet is a Model-Resource wrapper over the graph-node level RowSet.

### CODE REVIEW

File: jena-arq/src/main/java/org/apache/jena/riot/rowset/rw/RowSetReaderNone.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    @Override public RowSet read(InputStream in, Context context)       { IO.skipToEnd(in) ; return null; }
+
+    @Override public RowSet read(Reader in, Context context)            { IO.skipToEnd(in); return null; }
+
+    @Override public QueryExecResult readAny(InputStream in, Context context) { return null; }
+
+
+
+}
```
Comment (kinow) [inline jena-arq/src/main/java/org/apache/jena/riot/rowset/rw/RowSetReaderNone.java:54]: missing newline

File: jena-arq/src/main/java/org/apache/jena/riot/rowset/rw/RowSetWriterCSV.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        out.write(headerBytes);
+        if ( booleanResult )
+            out.write(yesBytes);
+        else
+            out.write(noBytes);
+        out.write(NL);
+    }
+
+    private static void output(AWriter out, RowSet rowSet) {
+            try {
```
Comment (kinow) [inline jena-arq/src/main/java/org/apache/jena/riot/rowset/rw/RowSetWriterCSV.java:74]: Wrong indentation.

File: jena-arq/src/main/java/org/apache/jena/riot/rowset/rw/RowSetWriterTSV.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    }
+
+    protected static NodeFormatter createNodeFormatter() {
+        // Use a Turtle formatter to format terms
+        return new NodeFormatterTTL(null, null);
+    }
+
+    private static final String NL   = "\n" ;
+    private static final String SEP  = "\t" ;
+            private static final String headerBytes = "?_askResult" + NL;
```
Comment (kinow) [inline jena-arq/src/main/java/org/apache/jena/riot/rowset/rw/RowSetWriterTSV.java:118]: Wrong indentation.

File: jena-arq/src/main/java/org/apache/jena/riot/rowset/rw/RowSetWriterThrift.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import org.apache.jena.riot.thrift.ThriftRDF;
+import org.apache.jena.sparql.exec.RowSet;
+import org.apache.jena.sparql.resultset.ResultSetException;
+import org.apache.jena.sparql.util.Context;
+
+public class RowSetWriterThrift implements RowSetWriter {
+
+    public static RowSetWriterFactory factory = lang -> {
+        if (!Objects.equals(lang, ResultSetLang.RS_Thrift ) )
+            throw new ResultSetException("RowSetWriter for RDF/Thift asked for a "+lang);
```
Comment (kinow) [inline jena-arq/src/main/java/org/apache/jena/riot/rowset/rw/RowSetWriterThrift.java:38]: s/Thift/Thrift

File: jena-arq/src/main/java/org/apache/jena/sparql/resultset/TSVInput.java [hunk 5]
```diff
@@ -18,66 +18,27 @@
...
-import org.apache.jena.sparql.engine.ResultSetStream ;
-import org.apache.jena.sparql.util.NodeFactoryExtra ;
+import org.apache.jena.riot.resultset.ResultSetLang;
+import org.apache.jena.riot.rowset.rw.RowSetReaderTSV;
+import org.apache.jena.sparql.exec.RowSet;
 
 /**
- * Input reader associated to {@link TSVOutput}.
+ * Input reader for Tab Separated Values format.
+ *  @deprecated To be removed
```
Comment (kinow) [inline jena-arq/src/main/java/org/apache/jena/sparql/resultset/TSVInput.java:31]: Extra space (nit-picking! <hides/>)

**Review by kinow — APPROVED** [review]
Found no blockers (not that I understand 100% of everything that changed). Few comments, but looks great! Thanks!

**Comment by afs** [comment]
Fixed - thanks!

**Pull request merged**